src/actions.ts

```diff
@@ -467,6 +475,75 @@ export const actions: {[k: string]: QueryHandler} = {
 			matches: await tables.users.selectAll(['userid', 'banstate'])`WHERE ip = ${res.ip}`,
 		};
 	},
+	// oauth is broken into a few parts
+	// oauth/page - public-facing part
+	// oauth/api/page - api part (does the actual action)
+	async 'oauth/authorize'(params) {
+		if (!params.redirect_uri) {
+			throw new ActionError("No redirect_uri provided");
+		}
+		const clientInfo = await getOAuthClient(params.client_id);
+
+		this.response.setHeader('Content-Type', 'text/html');
+		try {
+			let content = await fs.readFile(
```

Comment:
Looks like a constant that can be lifted out of this function. I forget if this project supports top-level await, if it followed my advice and removed unnecessary transpilation it does

Reply:
I do not believe it supports top-level await? It's only one read, though, so it could probably just be synchronous.

Reply:
Made it a constant.